### PR TITLE
Fixtests

### PR DESCRIPTION
**Pull request by mortoray**
Fixes some intermittendly failing tests.

NOTE: there's also a TODO comment added to Navigation. It's not really related, but i don't really want a PR just for it and I don't want to lose it either. 

This PR contains:
- [x] Documentation
- [x] Tests

### CODE REVIEW

File: Source/Fuse.Navigation/WhileCanGoBack.uno [hunk 1]
```diff
@@ -11,6 +11,8 @@ namespace Fuse.Navigation
 	public abstract class WhileHistoryTrigger : WhileTrigger
 	{
 		IBaseNavigation _context;
+		//TODO: This won't work due to being overidden in `OnUnrooted`. A good fix would be to remove this `public` part
+		//and alter the `Navigation` override to work with a `Node` (if that's even possible)
```
Comment (kusma) [inline Source/Fuse.Navigation/WhileCanGoBack.uno]: This seems unrelated to the rest of this patch...

File: Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno [hunk 2]
```diff
@@ -388,34 +388,34 @@ namespace Fuse.Reactive.Test
...
 		[Test]
-		[Ignore("https://github.com/fusetools/fuselibs-public/issues/107")]
 		public void MapTwoWayFloat3()
 		{
 			var p = new UX.Observable.MapTwoWayFloat3();
 			using (var root = TestRootPanel.CreateWithChild(p))
 			{
-				root.StepFrameJS();
+				//multi-step an Observable linked between two JavaScript elements involved
+				root.MultiStepFrameJS(2);
```
Comment (kusma) [inline Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno]: Maybe you could explain briefly in the commit message why two steps are needed?

File: Source/Fuse.Navigation/WhileCanGoBack.uno (hunk 1 above)
Comment (mortoray) [inline Source/Fuse.Navigation/WhileCanGoBack.uno]: Ah, was it a combined commit? CAn we leave this one for now. Reversion is unlikely since it's just a comment.

File: Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno (hunk 2 above)
Comment (mortoray) [inline Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno]: I assumed the extended comment on MultiStepFrameJS would be sufficient. :)

File: Source/Fuse.Navigation/WhileCanGoBack.uno (hunk 1 above)
Comment (kusma) [inline Source/Fuse.Navigation/WhileCanGoBack.uno]: I meant the other way around, so we'd keep the comment if we needed to revert the other part. But sure, it's not a big deal.

File: Source/Fuse.Common/Tests/FuseTest/TestRootNode.uno [hunk 3]
```diff
@@ -306,6 +306,22 @@ namespace FuseTest
 		}
 		
 		/**
+			If the test code involves multiple JavaScript elements that need to communicate between
+			each other, a single StepFrameJS may not be enough. This happens when an Observable in
+			one is connected to an Observable in another. The single StepFrameJS only takes care of
+			a single JS propagation, not the followup to the other element.
+			
+			This calls StepFrameJS multiple times to be correct in those situations.  THe number should
```
Comment (kusma) [inline Source/Fuse.Common/Tests/FuseTest/TestRootNode.uno]: `THe` -> `The`

File: Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno (hunk 2 above)
Comment (kusma) [inline Source/Fuse.Reactive.JavaScript/Tests/ObservableTest.uno]: OK, fair enough.

**Comment by mortoray** [comment]
@kusma Do you still need changes here, or can we merge this?

**Comment by kusma** [comment]
@mortoray: We should fix that typo before merging, at least. The other stuff is OK as-is.

**Comment by mortoray** [comment]
Yes, I thought I did, but I guess I didn't. Pushing now.
